### PR TITLE
fix: update dependencies and tests

### PR DESCRIPTION
**Pull request by ollz272**
Upgrades SQL Alchemy to support 1 and 2 series

Adds formal support for 3.11

### CODE REVIEW

**Comment by ollz272** [comment]
fixes https://github.com/yuval9313/FastApi-RESTful/issues/172

**Review by yuval9313 — CHANGES_REQUESTED** [review]

File: tests/test_inferring_router.py [hunk 1]
```diff
@@ -45,16 +45,6 @@ def test_infer_none(self, app: FastAPI, inferring_router: InferringRouter) -> No
...
-
-        @inferring_router.get("/noreturn-with-status", status_code=status.HTTP_204_NO_CONTENT)
-        def endpoint_noreturn_with_status_code() -> NoReturn:
-            pass
-
         app.include_router(inferring_router)
         openapi_spec = app.openapi()
         assert get_response_schema(openapi_spec, "/none") == {}
-        assert get_response_schema(openapi_spec, "/none-with-status", 204) is None
-        assert get_response_schema(openapi_spec, "/noreturn-with-status", 204) is None
```
[the remark this answers is not in the document]
Comment (ollz272) [reply tests/test_inferring_router.py:60]: https://github.com/tiangolo/fastapi/blob/master/fastapi/utils.py#LL36C1-L50C80
im not sure how long this has been the case, but when bumping fastapi this error occurred from these tests as a 204 is no longer allowed to return a body

[the remark this answers is not in the document]
Comment (yuval9313) [reply tests/test_inferring_router.py:60]: OK so your contribution is very much appreciated but I wanted to drop 3.6 alongside fastapi and take care better of sqlalchemy and more (like in the original fork) most of the changes already occurred on master (updated yesterday) so reset the poetry + workflow files and then attempt merge

On another subject in the latest versions of fastapi, they implemented an inferring router, so I dont think it is relevant anymore, thus deprecated. 

I've left the tests in until but I don't see how they are relevant anymore

File: pyproject.toml [hunk 2]
```diff
@@ -43,8 +43,8 @@ python = "^3.7"
 
 dataclasses = {version = "^0.8", python = "~3.6.2"}
 fastapi = "^0.89"
-pydantic = "^1.0,<2.0"
-sqlalchemy =  { version = "^1.4,<2.0", optional = true }
+pydantic = ">=1.10,<2.0"
```
Comment (ollz272) [inline pyproject.toml:46]: Theres an error in 1.9 in python 3.11, this bumps to a compatible version.

File: tests/test_inferring_router.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ollz272) [reply tests/test_inferring_router.py]: @yuval9313 pulled and updated

File: pyproject.toml (hunk 2 above)
Comment (yuval9313) [reply pyproject.toml:46]: You are right, however, out of backward compatibility, I don't want to force anyone to update (if for some reason they are using older pydantic). I thought I fixed it in master branch, but if it still causes issues I'll fix it as well

**Comment by ollz272** [comment]
@yuval9313 what changes are needed to get this merged?

**Comment by yuval9313** [comment]
Remove dependencies changes, I solved them generally

**Comment by ollz272** [comment]
> Remove dependencies changes, I solved them generally

Ok - i've updated the code itself. I guess you'll handle the package supporting 2.0 separately? We would like to upgrade our service to SQLAlchemy 2.0 and this is currently blocking

**Pull request closed without merging**